### PR TITLE
Enforcing JDK 11 on release

### PR DESCRIPTION
**Pull request by mkarg**
Enforcing the use of JDK 11 on release guarantees that the published Multi-Release JAR contains ALL code versions: JDK 8 (default fallback), JDK 9 (enhanced NIO) and JDK 10 (even further enhanced NIO). Without this enforcement, there would be a risk of not having some of these versions packed into the MRJAR due to the way the POM checks for the existence of supported JDK levels using JDK-enabled profiles.

This PR was requested by @olamy in https://github.com/codehaus-plexus/plexus-utils/pull/120#issuecomment-813327648, and is effective only in the `plexus-release` profile as requested by @rfscholte  in https://github.com/codehaus-plexus/plexus-utils/pull/148#discussion_r607186506. While it technically would be _enough_ to just enforce JDK 10, and while it would be _more future-proof_ to even enforce JDK 16, it follows @olamy 's preference mentioned in https://github.com/codehaus-plexus/plexus-utils/pull/120#issuecomment-813203371, hence enforce JDK 11 instead.

### CODE REVIEW

File: pom.xml
Comment (pzygielo) [inline pom.xml:235]: I propose to rename `id` and drop `-11`.
It would
- allow to update l. 242 in the future without the need to update `id` here,
- remove confusion, as currently any >= 11 is allowed, not only 11,
- follow the parent, where `enforce-maven` has no version specified.

File: pom.xml
Comment (rfscholte) [inline pom.xml:227]: If is called `plexus-release`, see https://github.com/codehaus-plexus/plexus-pom/blob/master/pom.xml#L670

File: pom.xml
Comment (mkarg) [reply pom.xml]: Done. Thank you for this proposal.

File: pom.xml
Comment (mkarg) [reply pom.xml]: Done. Good catch, thank you!

**Comment by mkarg** [comment]
@olamy @michael-o  Please review this PR.

I will keep it in draft state until you vote with +1 and I pretend to squash it before merging. :-)

File: pom.xml
Comment (rfscholte) [inline pom.xml:229]: `pluginManagement` doesn't make sense here. You want the plugins goal to be executed when this profile is activated

Comment (mkarg) [reply pom.xml:229]: `mvn help:effective-pom -Pplexus-release` proofs that the plugin's goal *actually is* executed when the profile is activated (thanks to POM inheritance), so I do not understand your claim. Why do you think it is *not*? Or do you want to say that it is *bad style* to simply re-configure inherited configurations? Or do you want to be safe against the case that the plugin *could* be removed from the parent, and *then* it would effectively not execute anymore?

Comment (mkarg) [reply pom.xml]: Done as per your request.

**Comment by olamy** [comment]
my understanding of draft vs not draft is draft means not ready for review vs ready for review.
I don't really understand why you are using such workflow

**Comment by mkarg** [comment]
My understanding of draft is "not ready for merge, but ready for review", so I did that to express that the code possibly will change due to the incoming change requests.

**Comment by olamy** [comment]
> My understanding of draft is "not ready for merge, but ready for review", so I did that to express that the code possibly will change due to the incoming change requests.

interesting point of view and workflow.
So what is non draft status? "ready for merge but too late for review" so nobody can make review anymore?

**Comment by mkarg** [comment]
non-draft means "ready to merge". Nothing else. Can still be reviewed. The sole point about "draft" mode is that it prevents clicking "merge" too early.

**Comment by mkarg** [comment]
@olamy @rfscholte This PR introduces the enforcement you requested. I answered your questions, and you did not further respond. So I would like to propose that you post a +1 when you want me to merge this one, or clearly request changes otherwise.

**Comment by slachiewicz** [comment]
+1 https://github.com/codehaus-plexus/plexus-pom/blob/master/pom.xml#L676

**Review by slachiewicz — APPROVED** [review]

**Comment by olamy** [comment]
please follow the usual workflow and remove draft status saying it's ready for review.
otherwise there is no need of a draft status...
We use that here and probably it's used in 99% of the github project.

File: pom.xml
Comment (MartinKanters) [inline pom.xml:243]: With the risk of being a PITA, I think the identing is a bit off compared to the rest of the file
```suggestion
                  <rules>
                    <requireJavaVersion>
                      <version>11</version>
                    </requireJavaVersion>
                  </rules>
```

**Comment by mkarg** [comment]
> please follow the usual workflow and remove draft status saying it's ready for review.
> otherwise there is no need of a draft status...
> We use that here and probably it's used in 99% of the github project.

In other projects I am contributing to, I was explicitly asked to *keep in draft* mode unless *no more* commits are expected to be put ontop (i. e. the committers are coaching the PR author towards a state that looks like all *would* be +1, hence the PR changes *heavily*). So please clearly tell me *how* that "usual" way is like. Currently I am still adding commits ontop as while the reviews run, change request *do* appear. Hence, *when* do committers know *when* to finally vote *if not* by the day it turns from draft to non-draft? So shall this really happen *outside* of draft mode? Just need to get a confirmation on that, and I will do that. Again, this is not that I want to propose another process or want to discuss yours, I just *need to know how you want it*. :-)

File: pom.xml
Comment (mkarg) [reply pom.xml:243]: Good catch! My IDE tricked me... Thanks for proposing this change!

**Comment by mkarg** [comment]
Fixed last change request, no more discussions unanswered, no -1 so far => switching from draft to non-draft now. Thanks to everybody for the feedback to the draft. Committers, please cast your VOTE now! :-)

**Review by slachiewicz — APPROVED** [review]

**Pull request merged**
